### PR TITLE
test: should throw original error message on non JSON response on large metadata

### PR DESCRIPTION
**Pull request by laljikanjareeya**
In recent PR it's failing.
Ref:
#1240 
#1238

### CODE REVIEW

File: system-test/storage.ts [hunk 1]
```diff
@@ -2144,19 +2144,18 @@ describe('storage', () => {
 
       const file = bucket.file('large-metadata-error-test');
       // Save file with metadata size more then 2MB.
-      await assert.rejects(
-        async () => {
-          await file.save('test', {
-            resumable: false,
+      await file.save(
```
Comment (stephenplusplus) [inline system-test/storage.ts:2147]: Let's not mix `await` with the callback usage below. You could use the `done` system, or try/catch.

**Review by stephenplusplus — CHANGES_REQUESTED** [review]

**Review by stephenplusplus — APPROVED** [review]

File: system-test/storage.ts (hunk 1 above)
Comment (stephenplusplus) [reply system-test/storage.ts]: Thanks!

**Review by JustinBeckwith — CHANGES_REQUESTED** [review]
Please use the `test: ` prefix for changes that don't affect users.  Anything with `fix: ` will trigger a release, which is not what we want here

File: system-test/storage.ts [hunk 2]
```diff
@@ -2143,22 +2143,19 @@ describe('storage', () => {
       };
 
       const file = bucket.file('large-metadata-error-test');
-      // Save file with metadata size more then 2MB.
```
Comment (JustinBeckwith) [inline system-test/storage.ts:2146]: Please switch this back to an `assert.rejects`.  When you try to do an assert in the catch, there's a chance the method just doesn't throw.  The `await assert.rejects` is careful about making sure both that it throws, and that it gets the right error message.

**Review by JustinBeckwith — CHANGES_REQUESTED** [review]

File: system-test/storage.ts (hunk 2 above)
Comment (stephenplusplus) [reply system-test/storage.ts:2146]: I was wondering about that. I saw the try/catch used in other tests, so we may have some more work to do across the libraries to be safe.

Comment (JustinBeckwith) [reply system-test/storage.ts:2146]: Yeah, I try to call em out when I see them.  Things were messy until `assert.rejects` made it's way consistently into node core.

**Comment by stephenplusplus** [comment]
> Anything with fix: will trigger a release, which is not what we want here

I agree this is a "test:" as opposed to a "fix:", but am confused about the repercussions of mis-labeling. I assume the downsides of unnecessary release PRs are potential resource exhaustion, in terms of GCP operations and CI overloading. However, I thought there was always a pending release PR. Is that not the case, they only get opened when the first fix/feat is merged? It seems like a narrow slice in time where we sit on master without having an auto-gen'd fix or dependency update, etc. So without knowing more, it sounds like it could be more trouble than its worth to not just let all commits trigger creating a release PR, if necessary?

I'm not heavily invested in either option, this just turned out to be a good time to learn more about how this works, what we're going for, and why.

**Comment by JustinBeckwith** [comment]
Yeah it's a really good question.  The release PR is only created when there are changes on master that would trigger a release - like a `fix` or `feat`.  By landing things with `fix`, it means a few things will happen:
- This will show up in the customer facing changelog that gets autogenerated.  Customers probably don't care that we fixed our own tests :) 
- It will open up the tracking PR

We are trying to keep the number of open PRs to a minimum where we can.  We also want to release bug fixes to developers as quickly as we can after them landing on master, so changes don't pile up.  As a result of both of these, it leads to us often merging release PRs with one or two fixes (a good thing!).  Cutting a bunch of releases that have no customer impacting changes is kinda not great, as it creates a lot of churn.

File: system-test/storage.ts (hunk 2 above)
Comment (laljikanjareeya) [reply system-test/storage.ts:2146]: @JustinBeckwith  Thanks for bringing this up.

File: system-test/storage.ts [hunk 3]
```diff
@@ -2155,9 +2154,7 @@ describe('storage', () => {
             },
           });
         },
-        {
-          message: 'Cannot parse response as JSON: Metadata part is too large.',
-        }
+        (err: Error) => err.message.indexOf('Metadata part is too large.') > -1
```
Comment (JustinBeckwith) [inline system-test/storage.ts:2157]: Should be able to simplify this entire line to:
```js
/Metadata part is too large/
```

That method takes an optional regex, or function to do the check.

**Review by stephenplusplus — APPROVED** [review]

File: system-test/storage.ts [hunk 4]
```diff
@@ -2143,22 +2143,16 @@ describe('storage', () => {
       };
 
       const file = bucket.file('large-metadata-error-test');
-      // Save file with metadata size more then 2MB.
-      await assert.rejects(
-        async () => {
-          await file.save('test', {
-            resumable: false,
+      await assert.rejects(async () => {
```
Comment (JustinBeckwith) [inline system-test/storage.ts:2146]: You can drop the `async` and `await here!  This is as easy as:
```js
await assert.rejects(
  file.save('test', {
    ....
  })
);
```

**Review by JustinBeckwith — APPROVED** [review]

**Pull request merged**